src/app/sidebar/components/CoursesList.tsx

```diff
@@ -15,18 +15,28 @@ type CoursesListProps = {
 
 export function CoursesList({ term, courses }: CoursesListProps): JSX.Element | null {
   const { subject } = useParams();
+  const calendarMatch = useMatch('/calendar/*');
+  const scheduleMatch = useMatch('/scheduler/*');
+
+  const createCard = (pid: string, code: string, subject: string, title: string) => {
+    if (calendarMatch)
+      return (
+        <LinkBox key={pid} as={Link} to={`/calendar/${term}/${subject}?pid=${pid}`} data-title={title}>
+          <Card title={title} subject={subject} code={code} />
+        </LinkBox>
+      );
+    else if (scheduleMatch) {
+      return <Card title={title} subject={subject} code={code} schedule={true} />;
+    }
+  };
 
   if (!courses || !courses[subject]) {
     return null;
   }
```

Comment:
This is SUPER nitty (and I don't exactly know if it is the case) but moving this above your `createCard` is likely a good idea. Since `createCard` is defined each time `CoursesList` is rendered, it is being defined when not being using which in this case is the `!courses || !courses[subject]`.